Filter sanity checks

jaisnan opened the pull request:
### Description of changes: 

Properties which are sanity checks needed to be removed from the UI. They are now marked with the property_class "sanity_check" which makes it easy to filter such checks out as the pipeline to add, remove and edit checks based on their property_class has already been setup.

### Resolved issues:

Resolves #1020 

### Call-outs:

Couldn't add tests for this UI change as it's a removal. In the future we should contain negative tests for similar changes.

### Testing:

* How is this change tested?

* Is this a refactor change?

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

danielsn commented on scripts/cbmc_json_parser.py:
```diff
@@ -373,19 +374,27 @@ def has_check_failure(properties, message):
+    return filter_properties(properties, None, "sanity_check")
 
-def filter_properties(properties, message):
+def filter_properties(properties, message=None, property_class=None):
```
Do we want a single `property_class` or a list?

jaisnan replied:
Maybe a wrapper function over this function which takes in a list of property_classes and messages? For this function itself, I think it's better to keep it simple, and call it over a list somewhere else.

celinval commented on scripts/cbmc_json_parser.py:
```diff
@@ -329,6 +329,7 @@ def postprocess_results(properties, extra_ptr_check):
     has_failed_unwinding_asserts = has_check_failure(properties, GlobalMessages.UNWINDING_ASSERT_DESC)
     has_reachable_undefined_functions = modify_undefined_function_checks(properties)
     properties, reach_checks = filter_reach_checks(properties)
+    properties = filter_sanity_checks(properties)[0]
```
Is there any chance this is empty?

celinval replied:
also, you should move this logic to be inside filter_sanity_checks

jaisnan replied:
sure, thanks!

zhassan-aws reviewed (changes requested):
We need to check if any of the sanity checks fail, and if so, report verification failure (and the sanity check that failed).

celinval commented on tests/kani/Invariants/fixme_niche_opt.rs:
```diff
@@ -1,38 +0,0 @@
-
-/// This testcase is currently failing with the following error:
-///
-/// [assertion.2] Reached assignment statement with unequal types Pointer { typ:
```
Good catch. As we discussed offline, sanity checks that fail should be visible and fail the verification. Thanks!

jaisnan replied:
let me address those cases and clean up some of the code as well

zhassan-aws commented on scripts/cbmc_json_parser.py:
```diff
@@ -369,23 +370,41 @@ def has_check_failure(properties, message):
+    return filtered_properties
 
-def filter_properties(properties, message):
+def filter_properties(properties, message=None, property_class=None):
```
The changes to this function are no longer needed, right?

jaisnan replied:
Yes, redundant changes.

jaisnan commented:
I've simplified this PR to just do the minimum and added a simple test to check for sanity_check. Will try to address comments on the spot to get it merged before tomorrow.

zhassan-aws reviewed (approved):
LGTM. In the future, we might want to make a sanity check failure more pronounced, e.g. through erroring out and not displaying any results at all. But we can do that in another PR.
